transport_nantes/topicblog/models.py

```diff
@@ -664,3 +664,121 @@ class TopicBlogEmailClicks(models.Model):
+        )
+
+    subject = models.CharField(max_length=80, blank=True)
+    header_image = models.ImageField(
+        upload_to='header/', blank=True,
+        help_text='résolution recommandée : 1600x500')
+
+    # Content #######################################################
+    body_text_1_md = models.TextField(blank=True)
+    body_image_1 = models.ImageField(
+        upload_to='body/', blank=True,
+        help_text='résolution recommandée : 1600x500')
+    body_image_1_alt_text = models.CharField(max_length=100, blank=True)
+
+    # Plus slug, template, title, comment, and social media fields,
+    # provided through abstract base class.
+
+
+    def get_absolute_url(self):
+        """Provide a link to view this object (by slug and id).
+        """
+        if self.slug:
+            return reverse("topicblog:view_press_by_pkid",
+                           kwargs={"pkid": self.pk,
+                                   "the_slug": self.slug})
+        else:
+            return reverse("topicblog:view_press_by_pkid_only",
+                           kwargs={"pkid": self.pk})
+
+    def get_edit_url(self):
+        """Provide a link to edit this object (by slug and id).
+        """
+        if not self.slug:
+            return reverse("topicblog:edit_press_by_pkid",
+                           kwargs={"pkid": self.pk})
+        else:
+            return reverse("topicblog:edit_press",
+                           kwargs={"pkid": self.pk,
+                                   "the_slug": self.slug})
+
+
+
+class TopicBlogPressSendRecord(models.Model):
+
+    """Represent the fact that we sent an press.
+    """
+    slug = models.SlugField(max_length=90, allow_unicode=True, blank=True)
+    mailinglist = models.ForeignKey(MailingList, on_delete=models.PROTECT)
+    recipient = models.ForeignKey(User, on_delete=models.PROTECT)
+    send_time = models.DateTimeField(auto_now=True)
+    # Open time is the time of the first instance of a beacon responding.
+    open_time = models.DateTimeField()
+    # Click time is the time of the first instance of a link being clicked.
+    click_time = models.DateTimeField()
+    # Unsubscribe time is the first instance of a user clicking the
+    # unsubscribe button, whether the unsubscribe is successful or
+    # not.  Note that we still have to send the user to the
+    # appropriate mailinglist unsubscribe page with user_id filled in
+    # (so that no press confirmation is required, which would be safer
+    # but would annoy most people).
+    unsubscribe_time = models.DateTimeField()
+
+
+class TopicBlogPressClicks(models.Model):
+
+    """Represent the fact that an press was clicked.
+
+    Note that TopicBlogPressSendRecord will record the first click,
+    but here we want to record all clicks that happen and where they
+    lead.
+
+    """
+    press = models.ForeignKey(TopicBlogPressSendRecord,
+                              on_delete=models.PROTECT)
+    click_time = models.DateTimeField()
```

Comment:
What's the difference in click_time in both classes ? 
The Record only register the first time while a TBPClicks is created on each click, is that right ?

Why saving specifically the first click in records while we can lookup the first occurrence with the FK ?

Reply:
I miss worded it, I meant why save the first click in the PressSendRecord and save a PressClick at the same time, while you could query Pressclick to get the first record existing for a given PressSendRecord. 

I guess it's to not have to query, but it could be a FK, for what it's worth.